File: LibraryApp/Application/Members/MemberService.cs
using Application.Members.DTOs;
using AutoMapper;
using Domain.Loans;
using Domain.Members;

namespace Application.Members;

public class MemberService : IMemberService
{
    private readonly IMapper _mapper;
    private readonly IMemberRepository _memberRepository;
    private readonly ILoanRepository _loanRepository;

    public MemberService(IMapper mapper, IMemberRepository memberRepository, ILoanRepository loanRepository)
    {
        _mapper = mapper;
        _memberRepository = memberRepository;
        _loanRepository = loanRepository;
    }

    public List<MemberDto> GetAll()
    {
        var members = _memberRepository.GetAll();
        return _mapper.Map<List<MemberDto>>(members);
    }

    public MemberDto GetById(int id)
    {
        var member = _memberRepository.GetById(id);
        return _mapper.Map<MemberDto>(member);
    }

    public void Insert(MemberCreateDto member)
    {
        var memberEntity = _mapper.Map<Member>(member);
        memberEntity.IsActive = true; 
        _memberRepository.Insert(memberEntity);
    }

    public void Update(MemberDto member)
    {
        var memberEntity = _mapper.Map<Member>(member);
        memberEntity.IsActive = true;
        _memberRepository.Update(memberEntity);
    }
    
    public void Delete(int id)
    {
        _memberRepository.Delete(id);
    }
}

File: angular/src/app/features/members/services/member.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { MemberDto, MemberCreateDto } from '../models/member.dto';
import { Observable } from 'rxjs';



@Injectable({
  providedIn: 'root'
})
export class MemberService {
    private api_url = `http://localhost:5147/api/members`;

    constructor(private http: HttpClient){}

    getAll(): Observable<MemberDto[]>{
        return this.http.get<MemberDto[]>(this.api_url);
    }

    getById(id: number): Observable<MemberDto>{
        return this.http.get<MemberDto>(`${this.api_url}/${id}`);
    }
 
    Insert(member: MemberCreateDto): Observable<MemberDto>{
        return this.http.post<MemberDto>(this.api_url, member);
    }

    Update(id: number, member: MemberDto): Observable<void>{
        return this.http.put<void>(`${this.api_url}/${id}`, member);
    }

    Delete(id: number): Observable<void>{
        return this.http.delete<void>(`${this.api_url}/${id}`);
    }
}
